### PR TITLE
Adding CTE style per Fishtown's Recommendations

### PR DESCRIPTION
**Pull request by lucas-smith**
Also, reorganized bullet points to make them feel more connected.

### CODE REVIEW

File: style/hudl_sql_style_guide.md [hunk 1]
```diff
@@ -45,9 +50,24 @@ description: A guide to writing clean, clear, and consistent SQL.
...
-* All dates should be formatted as 'YYYY-MM-DD'
-* Use `!=` for the inequality operator
+## `Common Table Expressions (CTEs)`
+
+* Use [Common Table Expressions](http://www.postgresql.org/docs/8.4/static/queries-with.html) often, and place them at the top
+* CTEs should be used to represent source tables to make code more readable
+* Only use full CTE Names when referencing them in the SELECT statement
+
+```sql
+WITH project_costs AS(
```
Comment (bogdan7900) [inline style/hudl_sql_style_guide.md]:
```suggestion
WITH project_costs AS (
```

File: style/hudl_sql_style_guide.md [hunk 2]
```diff
@@ -45,9 +50,24 @@ description: A guide to writing clean, clear, and consistent SQL.
...
   `SELECT ksr.backings AS b`
 
-* Use [Common Table Expressions](http://www.postgresql.org/docs/8.4/static/queries-with.html) (CTEs) early and often, and name them well.
-* All dates should be formatted as 'YYYY-MM-DD'
-* Use `!=` for the inequality operator
+## `Common Table Expressions (CTEs)`
+
+* Use [Common Table Expressions](http://www.postgresql.org/docs/8.4/static/queries-with.html) often, and place them at the top
+* CTEs should be used to represent source tables to make code more readable
+* Only use full CTE Names when referencing them in the SELECT statement
```
Comment (bogdan7900) [inline style/hudl_sql_style_guide.md]:
```suggestion
* Only use full CTE names when referencing them in the SELECT statement
```

Comment (bogdan7900) [reply style/hudl_sql_style_guide.md]: I don't see anything around the final SELECT statement. Do we want that to be 
```
SELECT 
  *
FROM [prior_CTE]
```
?

File: style/hudl_sql_style_guide.md [hunk 3]
```diff
@@ -45,9 +50,24 @@ description: A guide to writing clean, clear, and consistent SQL.
   __BAD__:
   `SELECT ksr.backings AS b`
 
-* Use [Common Table Expressions](http://www.postgresql.org/docs/8.4/static/queries-with.html) (CTEs) early and often, and name them well.
-* All dates should be formatted as 'YYYY-MM-DD'
-* Use `!=` for the inequality operator
+## `Common Table Expressions (CTEs)`
+
+* Use [Common Table Expressions](http://www.postgresql.org/docs/8.4/static/queries-with.html) often, and place them at the top
+* CTEs should be used to represent source tables to make code more readable
```
Comment (jaymewacha) [inline style/hudl_sql_style_guide.md]: is `source` the right word here?  or would `refs` make more sense?  To me, I think of the source tables only.

File: style/hudl_sql_style_guide.md [hunk 4]
```diff
@@ -45,9 +50,24 @@ description: A guide to writing clean, clear, and consistent SQL.
...
+* Only use full CTE names when referencing them in the SELECT statement
```
Comment (jaymewacha) [inline style/hudl_sql_style_guide.md]: maybe add `i.e. No aliases` so the reader knows what the full CTE names are meant to be used in place of

File: style/hudl_sql_style_guide.md (hunk 3 above)
Comment (lucas-smith) [reply style/hudl_sql_style_guide.md]: Per your suggestion added wording to make clearer.

File: style/hudl_sql_style_guide.md (hunk 4 above)
Comment (lucas-smith) [reply style/hudl_sql_style_guide.md]: Per your suggestion added wording to make clearer.

**Review by bogdan7900 — APPROVED** [review]
Looks good. Merge away!

**Pull request merged**